src/App.tsx

```diff
@@ -44,45 +47,59 @@ function App() {
     runContractCompilation(currentFileContent);
   }
 
-  const runContractCompilation = (currentFileContent: string) => {
-    fetch(cairoHostUrl, {
-      method: 'POST',
-      headers: {
-        accept: 'application/json',
-      },
-      body: JSON.stringify({
-        action: "compile-contract",
-        code: currentFileContent
-      })
-    })
-    .then(response => response.json())
-    .then(setContract)
-    .catch(setError)
-    .finally(() => setCompilingStatus(false));
+  const runContractCompilation = async (currentFileContent: string) => {
+    try {
+      const response = await fetch(cairoHostUrl, {
+        method: 'POST',
+        headers: {
+          accept: 'application/json',
+        },
+        body: JSON.stringify({
+          action: "compile-contract",
+          code: currentFileContent
+        })
+      });
+
+      const responseData = await response.json();
+      setCompilingStatus(false);
+      if (responseData.error) {
+        setCompilationError(responseData.error);
```

Comment:
Also, lets move this to a separate function and add annotation which will highlight the error in the code editor
Example
`  await remixClient.editor.addAnnotation({
          row: 18,
          column: 11,
          text: responseData.error,
          type: 'error'
        });
`